### PR TITLE
CAN 149:- Forgot Password UI and API Integration

### PR DESCRIPTION
**Pull request by ks-rahul**
Forgot Password UI and API Integration

### CODE REVIEW

File: src/components/ComponentPages/ForgotPassword/UI/index.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
...dv22uIEiLA4q7nvvCug8WTqzQveOH26fodo7g6uFe/a17W3+nFBAkRYENRdb1vkkz1CH9cPsVy/jrhr27PqMYvENYNlHAIesRiBYwRy0V+8iXP8+/fvX11Mr7L7ECueb/r48eMqm7FuI2BGWDEG8cm+7G3NEOfmdcTQw4h9/55lhm7DekRYKQPZF2ArbXTAyu4kDYB2YxUzwg0gi/41ztHnfQG26HbGel/crVrm7tNY+/1btkOEAZ2M05r4FB7r9GbAIdxaZYrHdOsgJ/wCEQY0J74TmOKnbxxT9n3FgGGWWsVdowHtjt9Nnvf7yQM2aZU/TIAIAxrw6dOnAWtZZcoEnBpNuTuObWMEiLAx1HY0ZQJEmHJ3HNvGCBBhY6jtaMoEiJB0Z29vL6ls58vxPcO8/zfrdo5qvKO+d3Fx8Wu8zf1dW4p/cPzLly/dtv9Ts/EbcvGAHhHyfBIhZ6NSiIBTo0LNNtScABFyNiqFCBChULMNNSdAhJyNSiECRCjUbEPNCRAhZ6NSiAARCjXbUHMCRMjZqBQiQIRCzTbUnAARcjYqhQgQoVCzDTUnQIScjUohAkQo1GxDzQkQIWejUogAEQo121BzAkTI2agUIkCEQs021JwAEXI2KoUIEKFQsw01J0CEnI1KIQJEKNRsQ80JECFno1KIABEKNdtQcwJEyNmoFCJAhELNNtScABFyNiqFCBChULMNNSdAhJyNSiECRCjUbEPNCRAhZ6NSiAARCjXbUHMCRMjZqBQiQIRCzTbUnAARcjYqhQgQoVCzDTUnQIScjUohAkQo1GxDzQkQIWejUogAEQo121BzAkTI2agUIkCEQs021JwAEXI2KoUIEKFQsw01J0CEnI1KIQJEKNRsQ80JECFno1KIABEKNdtQcwJEyNmoFCJAhELNNtScABFyNiqFCBChULMNNSdAhJyNSiECRCjUbEPNCRAhZ6NSiAARCjXbUHMCRMjZqBQiQIRCzTbUnAARcjYqhQgQoVCzDTUnQIScjUohAkQo1GxDzQkQIWejUogAEQo121BzAkTI2agUIkCEQs021JwAEXI2KoUIEKFQsw01J0CEnI1KIQJEKNRsQ80JECFno1KIABEKNdtQcwJEyNmoFCJAhELNNtScABFyNiqFCBChULMNNSdAhJyNSiEC/wGgKKC4YMA4TAAAAABJRU5ErkJggg=="
```
Comment (akuks) [inline src/components/ComponentPages/ForgotPassword/UI/index.tsx]: Can it be replaced by a variable?

File: src/components/ComponentPages/ForgotPassword/UI/index.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import styles from "../../Registration/UI/Registration.module.scss";
+
+import messages from "../../../../messages";
+
+const { Title, Text } = Typography;
+
+function ForgotPasswordUI({ form, onFinish, isModal, closeModal }) {
+  return (
+    <section className={styles.signup_wrapper + " " + styles.textCenter}>
+      <Form
```
Comment (akuks) [inline src/components/ComponentPages/ForgotPassword/UI/index.tsx:15]: Can we create a component and use Props to pass data. It looks similar to otp.tsx

File: src/components/ComponentPages/ForgotPassword/UI/otp.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import styles from "../../Registration/UI/Registration.module.scss";
+
+import messages from "../../../../messages";
+
+const { Title, Text } = Typography;
+
+export default function OTPVerify({ form, onFinish, isModal, closeModal }) {
+  return (
+    <section className={styles.signup_wrapper + " " + styles.textCenter}>
+      <Form
```
Comment (akuks) [inline src/components/ComponentPages/ForgotPassword/UI/otp.tsx]: Can we create a component and use Props to pass data. It looks similar to index.tsx under the ForgotPassword

File: src/components/ComponentPages/ForgotPassword/UI/password.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+import messages from "../../../../messages";
+import FormItem from "../../../common/formElements";
+
+const { Title } = Typography;
+
+function SetPasswordUI({ form, onFinish, isModal, closeModal }) {
+  return (
+    <section className={styles.signup_wrapper + " " + styles.textCenter}>
+      <Form
```
Comment (akuks) [inline src/components/ComponentPages/ForgotPassword/UI/password.tsx]: Not sure if it same as previous comments. If yes, then we can create a component and pass the props

File: src/components/ComponentPages/ForgotPassword/index.tsx [hunk 5]
```diff
@@ -1,5 +1,109 @@
...
+    };
+
+    let res = await forgotPasswordUpdate(body);
+
+    if (res && res.status_code === 200) {
+      passwordForm.resetFields();
+      setIsScreen(0);
+    }
+
+    if (isModal) {
```
Comment (akuks) [inline src/components/ComponentPages/ForgotPassword/index.tsx]: Can we replace with ternary operator?

File: src/components/ComponentPages/Login/index.tsx [hunk 6]
```diff
@@ -4,39 +4,42 @@ import { useRouter } from "next/router";
...
+    if (values.remember) {
+      localStorage.setItem(
+        "rememberme",
+        JSON.stringify({
+          username: values.username,
+          password: values.password,
+        })
+      );
+    }
+    // try {
```
Comment (akuks) [inline src/components/ComponentPages/Login/index.tsx]: Remove this code if not required

File: src/components/ComponentPages/Login/loginModal.tsx [hunk 7]
```diff
@@ -11,6 +11,7 @@ const LoginModal = () => {
   return (
     <Fragment>
       <Modal
+        style={{ top: "20px" }}
         visible={visible}
         // title="Title"
```
Comment (akuks) [inline src/components/ComponentPages/Login/loginModal.tsx]: Remove the commented code if not required

File: src/components/ComponentPages/NickName/NickNameUI/index.tsx [hunk 8]
```diff
@@ -25,35 +25,34 @@ export default function NickNameUI({
...
+  handleNickNameCancel,
 }) {
-
   const [data, setData] = useState(originData);
   const columns = [
     {
-      title: 'Sr',
-      dataIndex: 'srNo',
-      width: '5%',
+      title: "Sr",
```
Comment (akuks) [inline src/components/ComponentPages/NickName/NickNameUI/index.tsx:33]: Not sure why `'` replaced with `"`. JS treats `'` faster than `"`. Same for all

File: src/components/common/socialLogin/social-login.module.scss [hunk 9]
```diff
@@ -1,5 +1,5 @@
 .wrapper {
-  display: none;
+  // display: none;
```
Comment (akuks) [inline src/components/common/socialLogin/social-login.module.scss]: Remove the commented if not required

**Review by akuks — CHANGES_REQUESTED** [review]
Please do the required changes

File: src/components/ComponentPages/ForgotPassword/UI/index.tsx (hunk 1 above)
Comment (ks-rahul) [reply src/components/ComponentPages/ForgotPassword/UI/index.tsx]: replaced

File: src/components/ComponentPages/ForgotPassword/UI/index.tsx (hunk 2 above)
Comment (ks-rahul) [reply src/components/ComponentPages/ForgotPassword/UI/index.tsx:15]: forgot password components changed to one component and manage by props

File: src/components/ComponentPages/ForgotPassword/UI/otp.tsx (hunk 3 above)
Comment (ks-rahul) [reply src/components/ComponentPages/ForgotPassword/UI/otp.tsx]: done

File: src/components/ComponentPages/ForgotPassword/UI/password.tsx (hunk 4 above)
Comment (ks-rahul) [reply src/components/ComponentPages/ForgotPassword/UI/password.tsx]: done

File: src/components/ComponentPages/ForgotPassword/index.tsx (hunk 5 above)
Comment (ks-rahul) [reply src/components/ComponentPages/ForgotPassword/index.tsx]: changed

File: src/components/ComponentPages/Login/index.tsx (hunk 6 above)
Comment (ks-rahul) [reply src/components/ComponentPages/Login/index.tsx]: removed

File: src/components/ComponentPages/Login/loginModal.tsx (hunk 7 above)
Comment (ks-rahul) [reply src/components/ComponentPages/Login/loginModal.tsx]: removed

File: src/components/common/socialLogin/social-login.module.scss (hunk 9 above)
Comment (ks-rahul) [reply src/components/common/socialLogin/social-login.module.scss]: Done

File: src/components/ComponentPages/NickName/NickNameUI/index.tsx (hunk 8 above)
Comment (ks-rahul) [reply src/components/ComponentPages/NickName/NickNameUI/index.tsx:33]: it's changed by formatter.

**Review by akuks — APPROVED** [review]
All good. All test cases passed.

**Pull request merged**